warehouse/query-core/src/test/java/datawave/query/iterator/SourceManagerTest.java

```diff
@@ -51,6 +55,112 @@ public void setup() throws ParseException, IOException {
         Map<String,String> options = Maps.newHashMap();
         
         counter.init(data, options, new MockIteratorEnvironment());
+        
+        dataIterator = new SortedListKeyValueIterator(createTestData());
+    }
+    
+    @Test
+    public void dataIntegrityTest() throws IOException {
+        SourceManager manager = new SourceManager(dataIterator);
+        manager.setInitialSize(1);
+        
+        // pre-seek both iterators
+        SortedKeyValueIterator<Key,Value> copy1 = manager.deepCopy(null);
+        copy1.seek(new Range(), Collections.emptyList(), false);
+        SortedKeyValueIterator<Key,Value> copy2 = manager.deepCopy(null);
+        copy2.seek(new Range(), Collections.emptyList(), false);
+        
+        // individual next loops
+        int copy1Count = 0;
+        while (copy1.hasTop()) {
+            assertNotEquals("unexpected topKey on iteration=" + copy1Count, null, copy1.getTopKey());
+            copy1Count++;
+            copy1.next();
+        }
+        
+        int copy2Count = 0;
+        while (copy2.hasTop()) {
+            assertNotEquals("unexpected topKey on iteration=" + copy2Count, null, copy2.getTopKey());
+            copy2Count++;
+            copy2.next();
+        }
+        
+        assertTrue("both copies should have the same number of next calls; copy1=" + copy1Count + " copy2=" + copy2Count, copy1Count == copy2Count);
+        
+        // re-seek
+        copy1.seek(new Range(), Collections.emptyList(), false);
+        copy2.seek(new Range(), Collections.emptyList(), false);
+        
+        // alternating next loops
+        int alternatingCount = 0;
+        while (copy1.hasTop() && copy2.hasTop()) {
+            assertTrue(copy1.getTopKey().equals(copy2.getTopKey()));
+            alternatingCount++;
+            copy1.next();
+            copy2.next();
+        }
+        
+        assertFalse(copy1.hasTop());
+        assertFalse(copy2.hasTop());
+        assertEquals(copy1Count, alternatingCount);
+        
+        // different ranges
+        copy1.seek(new Range(new Key("20121126_2"), true, new Key("20121126_3"), false), Collections.emptyList(), false);
```

Comment:
I would prefer to see this test broken out into the cases of "same range" - "different range". If a future change would only break one behavior it will be obvious up front which behavior is broken.

Reply:
done